src/utils/web3-provider/methods/utils.js

```diff
@@ -27,45 +31,33 @@ const getSanitizedTx = tx => {
 const setEvents = (promiObj, tx, dispatch) => {
   // create a no reference copy specifically for notification
   const newTxObj = _.clone(tx);
-  newTxObj['data'] = newTxObj.hasOwnProperty('input')
-    ? newTxObj.input
-    : newTxObj.data;
-  newTxObj.date = new Date().getTime();
-  const isExcempt = newTxObj.hasOwnProperty('handleNotification');
-  delete newTxObj['r'];
-  delete newTxObj['v'];
-  delete newTxObj['s'];
-  delete newTxObj['chainId'];
-  delete newTxObj['input'];
-  delete newTxObj['hash'];
-  delete newTxObj['__typename'];
+  newTxObj.type = notificationTypes.out;
+  const isExempt = newTxObj.hasOwnProperty('handleNotification');
 
   promiObj
     .once('transactionHash', hash => {
-      newTxObj.status = 'PENDING';
+      newTxObj.status = txTypes.pending;
       newTxObj.transactionHash = hash;
-      if (!isExcempt) {
-        const notification = new Notification(newTxObj);
+      if (!isExempt) {
+        const notification = new Notification(newTxObj, true);
         dispatch('notifications/addNotification', notification, {
           root: true
         });
       }
     })
     .on('receipt', res => {
       newTxObj.transactionHash = res.transactionHash;
-      newTxObj.status = 'SUCCESS';
-      if (!isExcempt) {
-        const notification = new Notification(newTxObj);
-        dispatch('notifications/updateNotification', notification, {
-          root: true
-        });
-      }
+      newTxObj.status = txTypes.success;
```

Comment:
@gamalielhere i removed isExempt here because if the notification has a handler, it will never update the notification...and i think it might have it since it is just an update...not sure...or it seems like it may be a new obj every time also @_@